### PR TITLE
[PM-11764] Implement account switching and sdk initialization

### PR DESCRIPTION
**Pull request by coroiu**
## 📔 Objective

- Expose crypto client
- Add error handling

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .editorconfig
Comment (Hinton) [inline .editorconfig:1]: These changes seems unrelated? And we probably want to change quite_type for ts ...

File: crates/bitwarden-wasm-internal/src/crypto.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+impl ClientCrypto {
+    /// Initialization method for the user crypto. Needs to be called before any other crypto
+    /// operations.
+    pub async fn initialize_user_crypto(&self, req: InitUserCryptoRequest) -> Result<()> {
+        Ok(self
+            .0
+            .crypto()
+            .initialize_user_crypto(req)
+            .await
+            .map_err(bitwarden_core::Error::EncryptionSettings)?)
```
Comment (Hinton) [inline crates/bitwarden-wasm-internal/src/crypto.rs:24]: **suggestion:** If we add bitwarden_core error as transparent #from in this crates error, you should be able to avoid the explicit error conversion and rely on the `thiserror` macro from implementation.

This is the pattern I updated uniffi to use in https://github.com/bitwarden/sdk/pull/1124.

File: crates/bitwarden-crypto/src/enc_string/asymmetric.rs [hunk 2]
```diff
@@ -11,11 +11,16 @@ use crate::{
...
-
 // This module is a workaround to avoid deprecated warnings that come from the ZeroizeOnDrop
 // macro expansion
 #[allow(deprecated)]
 mod internal {
+    #[cfg(feature = "wasm")]
+    #[wasm_bindgen::prelude::wasm_bindgen(typescript_custom_section)]
+    const TS_CUSTOM_TYPES: &'static str = r#"
+    export type AsymmetricEncString = string;
+    "#;
```
Comment (Hinton) [inline crates/bitwarden-crypto/src/enc_string/asymmetric.rs:22]: We have a `uniffi_support.rs` file, would it make sense to add something similar for these custom logics? Or should we encourage lifting in the uniffi logic to where it's needed.

Comment (coroiu) [reply crates/bitwarden-crypto/src/enc_string/asymmetric.rs:22]: I added `custom_types.rs` which is inspired by that uniffi file, but I think adding the types where they are defined is a preferred approach. It's just that we sometimes use external type and so I needed both approaches

File: .editorconfig
Comment (coroiu) [reply .editorconfig:1]: Oh that's why I suddenly got lots of prettier errors, lol, yeah ok I'll split this out

File: crates/bitwarden-wasm-internal/src/crypto.rs (hunk 1 above)
Comment (coroiu) [reply crates/bitwarden-wasm-internal/src/crypto.rs]: I have chosen another path which simply converts all errors to strings. I'll make sure we have a follow-up ticket to investigate the best method for catching errors in JS

**Comment by coroiu** [comment]
Adding @dani-garcia as reviewer because @Hinton is traveling/at conference

File: crates/bitwarden-core/src/mobile/crypto.rs [hunk 3]
```diff
@@ -6,6 +6,8 @@ use bitwarden_crypto::{
 };
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
+#[cfg(feature = "wasm")]
+use {tsify_next::Tsify, wasm_bindgen::prelude::*};
```
Comment (dani-garcia) [inline crates/bitwarden-core/src/mobile/crypto.rs:10]: Huh, I just discovered now thanks to this that you can use a top level `{}` block in a `use` statement, neat!

File: crates/bitwarden-wasm-internal/src/client.rs [hunk 4]
```diff
@@ -27,7 +29,7 @@ fn convert_level(level: LogLevel) -> Level {
 // Rc<...> is to avoid needing to take ownership of the Client during our async run_command
 // function https://github.com/rustwasm/wasm-bindgen/issues/2195#issuecomment-799588401
 #[wasm_bindgen]
-pub struct BitwardenClient(Rc<Client>);
+pub struct BitwardenClient(pub(crate) Rc<Client>);
```
Comment (dani-garcia) [inline crates/bitwarden-wasm-internal/src/client.rs:32]:
```suggestion
pub struct BitwardenClient(Rc<Client>);
```

Nit: This inner value is only used on it's own impl as far as I know, so we don't need it to be `pub(crate)`

**Review by dani-garcia — DISMISSED** [review]
LGTM!

**Review by dani-garcia — APPROVED** [review]
